packages/ckeditor5-dev-env/lib/release-tools/tasks/update-package-versions.js

```diff
@@ -0,0 +1,150 @@
+
+			if ( pathToUpdate.commit ) {
+				pathsToCommit.push( pathToUpdate.path );
+			}
+		}
+	}
+
+	if ( options.dryRun ) {
+		console.log( chalk.yellow( 'DRY RUN mode - displaying changes instead of committing.' ) );
+
+		// TODO: Changes will be displayed here
+		console.log( totalResult );
+	} else if ( pathsToCommit.length ) {
+		console.log( '\n📍 ' + chalk.blue( 'Committing the changes...\n' ) );
+
+		for ( const path of pathsToCommit ) {
+			const execOptions = {
+				stdio: 'inherit',
+				cwd: path
+			};
+
+			console.log( chalk.blue( path ) );
+			execSync( `git add ${ path }`, execOptions );
+			execSync( 'git commit -m "Internal: Updated all CKEditor 5 dependencies ' +
+				'in `packages/*` to the latest version. [skip ci]"', execOptions );
+		}
+
+		console.log( '\n📍 ' + chalk.green( `Successfully committed ${ pathsToCommit.length } files!\n` ) );
+	}
+
+	if ( totalResult.updated ) {
+		console.log( '\n📍 ' + chalk.green( `Updated total of ${ totalResult.updated } files!\n` ) );
+	} else {
+		console.log( '\n📍 ' + chalk.green( 'No files needed an update.\n' ) );
+	}
+};
+
+/**
+ * Updates `@ckeditor/ckeditor5-*` and `ckeditor5` dependencies in the specified directory to the latest version.
+ * Latest version is taken from the `version` property from the root of the `package.json` file, since that value
+ * should already have been bumped at this point of release process.
+ *
+ * @param {String} pathToUpdate directory containing files to update
+ * @param {String} cwd The root directory of the repository in which
+ * the script is being used.
+ * @returns {Object} Object containing numerical values counting files found and updated, as well as array of all changes made.
+ */
+function updateDirectory( pathToUpdate, cwd, dryRun ) {
+	const globPattern = pathToUpdate + '/*/package.json';
+	const packageJsonArray = glob.sync( globPattern, { cwd } );
+
+	if ( !packageJsonArray.length ) {
+		return { found: 0, updated: 0 };
+	}
+
+	let updatedFiles = 0;
+	const differences = [];
+
+	for ( const file of packageJsonArray ) {
+		const currentFileData = fs.readFileSync( file, 'utf-8' );
+		const parsedData = JSON.parse( currentFileData );
+		const version = parsedData.version;
+
+		// Update only the CKEditor 5 dependencies, except the *-dev and *-inspector.
+		const regex = /^@ckeditor\/ckeditor5-(?!dev|inspector)|^ckeditor5$/;
+
+		for ( const dependency in parsedData.dependencies ) {
+			if ( !regex.test( dependency ) ) {
+				continue;
+			}
+
+			parsedData.dependencies[ dependency ] = `^${ version }`;
+		}
+
+		for ( const dependency in parsedData.devDependencies ) {
+			if ( !regex.test( dependency ) ) {
+				continue;
+			}
+
+			parsedData.devDependencies[ dependency ] = `^${ version }`;
+		}
+
+		const newFileData = JSON.stringify( parsedData, null, 2 ) + '\n';
+
+		if ( currentFileData !== newFileData ) {
+			updatedFiles++;
+			differences.push( diff( currentFileData, newFileData ) );
```

Comment:
Currently, the diff in your script is needed only if `--dry-run` flag is set, so it does not need to be calculated always.

Or maybe do we want to display the changes always, even after committing them?

Reply:
I guess there is no reason for displaying the diff when executing the script in normal mode (without the `--dry-run` option).